core/internal/flux_monitor_simulated_blockchain_test.go

```diff
@@ -0,0 +1,370 @@
+	time.Sleep(time.Duration((waitTimeMs + waitTimeMs/20) * int64(time.Millisecond)))
+	f.aggregatorContractAddress, _, f.aggregatorContract, err =
+		faw.DeployFluxAggregator(f.sergey, f.backend,
+			linkAddress, big.NewInt(paymentAmount), timeout, decimals, description)
+	f.backend.Commit() // Must commit contract to chain before we can fund with LINK
+	require.NoError(t, err,
+		"failed to deploy FluxAggregator contract to simulated ethereum blockchain")
+	_, err = f.linkContract.Transfer(f.sergey, f.aggregatorContractAddress,
+		oneEth) // Actually, LINK
+	require.NoError(t, err, "failed to fund FluxAggregator contract with LINK")
+	_, err = f.aggregatorContract.UpdateAvailableFunds(f.sergey)
+	require.NoError(t, err, "failed to update aggregator's vailableFunds field")
+	f.backend.Commit()
+	availableFunds, err := f.aggregatorContract.AvailableFunds(nil)
+	require.NoError(t, err, "failed to retrieve AvailableFunds")
+	require.Equal(t, availableFunds, oneEth)
+	ilogs, err := f.aggregatorContract.FilterAvailableFundsUpdated(nil,
+		[]*big.Int{oneEth})
+	require.NoError(t, err, "failed to gather AvailableFundsUpdated logs")
+	logs := cltest.GetLogs(t, nil, ilogs)
+	require.Len(t, logs, 1, "a single AvailableFundsUpdated log should be emitted")
+
+	// Add the participating oracles. Ends up with minAnswers=restartDelay=2,
+	// maxAnswers=3
+	oracleList := []common.Address{f.neil.From, f.ned.From, f.nallory.From}
+	for numOracles, o := range oracleList {
+		n := uint32(numOracles)
+		_, err = f.aggregatorContract.AddOracle(f.sergey, o, o, n, n+1, n)
+		require.NoError(t, err, "failed to update oracles list")
+	}
+	f.backend.Commit()
+	iaddedLogs, err := f.aggregatorContract.FilterOracleAdded(nil, oracleList)
+	require.NoError(t, err, "failed to gather OracleAdded logs")
+	addedLogs := cltest.GetLogs(t, nil, iaddedLogs)
+	require.Len(t, addedLogs, len(oracleList), "should have log for each oracle")
+	iadminLogs, err := f.aggregatorContract.FilterOracleAdminUpdated(nil,
+		oracleList, oracleList)
+	require.NoError(t, err, "failed to gather OracleAdminUpdated logs")
+	adminLogs := cltest.GetLogs(t, nil, iadminLogs)
+	require.Len(t, adminLogs, len(oracleList), "should have log for each oracle")
+	for oracleIdx, oracle := range oracleList {
+		require.Equal(t, oracle,
+			addedLogs[oracleIdx].(*faw.FluxAggregatorOracleAdded).Oracle,
+			"log for wrong oracle emitted")
+		require.Equal(t, oracle,
+			adminLogs[oracleIdx].(*faw.FluxAggregatorOracleAdminUpdated).Oracle,
+			"log for wrong oracle emitted")
+	}
+	return f
+}
+
+type answerParams struct {
+	fa                          *fluxAggregator
+	roundId, answer             int64
+	from                        *bind.TransactOpts
+	isNewRound, completesAnswer bool
+}
+
+// checkUpdateAnswer verifies all the logs emitted by fa's FluxAggregator
+// contract after an updateAnswer with the given values.
+func checkUpdateAnswer(t *testing.T, p answerParams,
+	currentBalance int64, receiptBlock uint64) {
```

Comment:
Might want to add a call to `t.Helper()` at the top of this method.

Reply:
I don't really understand `t.Helper`'s purpose. Why would I want to elide information from test failure output?

Reply:
I added this, although I still don't really understand its purpose.